ViewEnvironmentUI/Sources/ViewEnvironmentPropagating.swift

```diff
@@ -272,6 +272,20 @@ extension ViewEnvironmentPropagating {
             objc_setAssociatedObject(self, &AssociatedKeys.descendantsOverride, newValue, .OBJC_ASSOCIATION_RETAIN)
         }
     }
+
+    @_spi(ViewEnvironmentWiring)
+    public var environmentAncestorTreeHash: AnyHashable {
```

Comment:
Maybe this is just for the prototype; but the hash isn't guaranteed to be different if the tree is different. I'd throw this into something like:

```
struct Ancestors : Equatable {

  private var IDs : [ObjectIdentifier]

  init(node : ViewController) {
    IDs = sequence(first: first, next: \.environmentAncestor).map {
      ObjectIdentifier($0)
    }
  }
}
```

Reply:
`ObjectIdentifier` is what I was looking for, thanks!
But yeah, I knew this wasn't quite right.